### PR TITLE
[Reland][ONNX] Move all torch.onnx.export related tests to test/onnx

### LINKED ISSUE #87320: [ONNX] Move all torch.onnx.export related tests into test/onnx
Some CI tests on `test.onnx.export` are scattered under test/jit, and test/jit doesn't share the same testing environment as test/onnx which might create the discrepancy. Most of them can be categorized into test/onnx/test_pytorch_onnx/no_runtime.py.

### PR DESCRIPTION
**Pull request by titaiwangms**
Stack from [ghstack](https://github.com/ezyang/ghstack) (oldest at bottom):
* __->__ #87292

Moving torch.onnx.export related tests to test/onnx integrates ONNX tests to the same CI machine, so the testing environment can be better managed.

Fixes https://github.com/pytorch/pytorch/issues/87320

### CODE REVIEW

File: test/onnx/test_pytorch_onnx_no_runtime.py [hunk 1]
```diff
@@ -427,7 +428,7 @@ def forward(self, x):
         onnx_model = export_to_onnx(
             MyClip(),
             torch.randn(3, 4, requires_grad=True),
-            custom_ops=[common_utils.custom_op("aten::clamp", bad_clamp, 9)],
+            custom_ops=[common_utils.custom_op("aten::clamp", bad_clamp, GLOBALS.export_onnx_opset_version)],
```
Comment (AllenTiTaiWang) [inline test/onnx/test_pytorch_onnx_no_runtime.py:431]: This test was broken for a while. I wonder why CI didn't catch this.

File: test/onnx/test_export_modes.py [hunk 2]
```diff
@@ -66,10 +61,10 @@ def foo(a):
             return (a, a)
         f = io.BytesIO()
         x = torch.ones(3)
-        torch.onnx._export(foo, (x,), f)
```
Comment (AllenTiTaiWang) [inline test/onnx/test_export_modes.py:69]: in `torch.onnx._export`, it has a parameter called export_type that is not exposed in `torch.onnx.export`, but we have its tests right above blocks. I am thinking if we should delete those as it's not exposed anyway?

File: test/onnx/test_pytorch_onnx_no_runtime.py (hunk 1 above)
Comment (AllenTiTaiWang) [reply test/onnx/test_pytorch_onnx_no_runtime.py]: Found the current CI only has Caffe2 included version, so the tests with `skipIfCaffe2` is skipped. Those tests are all aten_fallback related.

Comment (thiagocrepaldi) [reply test/onnx/test_pytorch_onnx_no_runtime.py]: We need an issue for that. ONNX Runtime heavily depends on BUILD_CAFFE2=0 builds to properly work and PyTorch CI is not guarding those scenarios

File: test/onnx/test_export_modes.py (hunk 2 above)
Comment (thiagocrepaldi) [reply test/onnx/test_export_modes.py:69]: That should be fine. We don't need to export to all formats. ONNX uses protobuf anyway

**Review by thiagocrepaldi — APPROVED** [review]

**Comment by AllenTiTaiWang** [comment]
@malfet please take a look. It's moving tests cases.

File: test/quantization/eager/test_quantize_eager_ptq.py [hunk 3]
```diff
@@ -1444,26 +1443,6 @@ def forward(self, indices, offsets, linear_in):
         self.assertTrue('DynamicQuantizedLinear' in str(q_model))
 
 class TestQuantizeEagerONNXExport(JitTestCase):
```
Comment (BowenBao) [inline test/quantization/eager/test_quantize_eager_ptq.py:1445]: this class has onnx name in there somehow..

**Review by BowenBao — APPROVED** [review]

**Comment by BowenBao** [comment]
@kit1980 I think this likely is out of onnx merge rules. Please take a look.

**Review by kit1980 — APPROVED** [review]

File: test/quantization/eager/test_quantize_eager_ptq.py (hunk 3 above)
Comment (malfet) [reply test/quantization/eager/test_quantize_eager_ptq.py:1445]: Yes, but other teams are using it, we should coordinate the name change with them

**Comment by AllenTiTaiWang** [comment]
> What should users do if they encounter errors (AttributeError: 'TestQuantizeEagerONNXExport' object has no attribute '_test_lower_graph_impl' ) due to the following deletion? Would this be breaking whenever "self._test_lower_graph_impl(model, data)" when self is TestQuantizeEagerONNXExport?
> 
> image

I can't see internal test failing. The goal of here is to move torch.onnx.export tests into test/onnx, so that we can have our API tested on the same environment. Is there a way you can do that in these specific tests? Or anything I can do?

**Comment by weiwangmeta** [comment]
> test/quantization/eager/test_quantize_eager_ptq.py test_lower_graph_linear and test_lower_graph_conv2d

Can you please test the above two cases? If they pass, then internal should be able to pass. You may need to remove the two skips (FBGEMM and CAFFE2) to trigger the two cases. 
Details below: 
https://github.com/pytorch/pytorch/blob/master/test/quantization/eager/test_quantize_eager_ptq.py#L1468-L1482

**Review by malfet — APPROVED** [review]

**Comment by malfet** [comment]
Thank you for the follow up, looks good to me

**Comment by AllenTiTaiWang** [comment]
> Thank you for the follow up, looks good to me (though ghstack for a single PR feels a bit excessive)

Thanks for the help.

**Pull request closed without merging**